### ISSUE TITLE
When the rate limit is hit, realizing the promise returned by create-message! gives an ex-info containing retry information

### ISSUE THREAD

**Issue by pmonks** [IGJoshua/discljord#61]
**Describe the bug**
In the spirit of "failing fast", I've wrapped some of discljord's APIs so that if an API call returns an `ex-info` (indicating an error was returned from Discord), that `ex-info` is thrown which ultimately causes it to be logged, so that I have visibility into what's going on.  Previously my bot ignored these error responses which meant I had no idea when Discord errors were occurring.  [You can see the current WIP of this approach here.](https://github.com/pmonks/futbot/blob/dev/src/futbot/message_util.clj#L23-L29)

This is all working well, **except** that one class of `ex-info` that I'm now seeing returned when I realise the promise relates to rate limit errors returned by Discord.  Here's an example `ex-data` from such an `ex-info`:

```
{:global false, :message You are being rate limited., :retry-after 793}
```

This is unexpected, because discljord automatically retries in these cases (and I can see that happening), so I wouldn't have expected to be notified (via an `ex-info` return value) that the rate limit had been exceeded.

**Version Information**
 - Clojure Version: 1.10.1
 - Discljord Version: 1.1.1
 - JVM Version: observed on both AdoptOpenJDK JVM 11.0.9+11 (x86_64) and Oracle Corporation JVM 11.0.9+11 (amd64)
 - OS bot is hosted on: macOS (dev) and Linux (production) - the issue doesn't appear to be OS specific

**To Reproduce**
1. Have a bot repeatedly send messages to Discord to the point that it exceeds the rate limit
2. Ensure the bot realises each call to `create-message!`
3. Log or print any realised responses that are an `ex-info`

Minimal reproduction case:
Start a REPL, with discljord available:
```
$ clj -Sdeps '{:deps {org.suskalo/discljord {:mvn/version "1.1.1"}}}'
Clojure 1.10.1
```
Copy & paste the following mini-bot, replacing the token and channel snowflake values as appropriate:
```clojure
(require '[discljord.messaging :as m])

(def token "YOUR_DISCORD_TOKEN")
(def channel-id "SNOWFLAKE_OF_TEST_CHANNEL_YOU_CAN_SPAM_WITHOUT_ANNOYING_ANYONE")

(let [message-ch (m/start-connection! token)]
  (try
    (doall
      (for [x (range 100)]
        (let [response @(m/create-message! message-ch channel-id :content "Rate limit testing - apologies for spam")]
          (if (= (class response) clojure.lang.ExceptionInfo)
            (println "ℹ️ got an ex-info:" (ex-data response))))))
    (finally
      (m/stop-connection! message-ch))))
```

Note: in my limited testing this mini-bot doesn't seem to reproduce the issue, though I can't tell if that's because it's not hitting the rate-limit, or because the issue is subtler than this simplistic example can reproduce.

**Expected behavior**
The mini-bot runs without printing any retry-specific `ex-info`s (though there may be other `ex-info`s printed, if the Discord API is in struggletown when you run it).

**Logs**
N/A

**Additional context**
N/A

**Comment by IGJoshua** [comment]
@pmonks This should be fixed as of 1.2.0, if you could confirm the fix, that would be appreciated!

**Comment by pmonks** [comment]
I've been running the 1.2.0-RC since it was released (a week or so, IIRC?) and haven't seen this issue once, so I'm happy to close this issue (it being impossible to prove a negative, yadda yadda).

I'll also upgrade to 1.2.0 final and keep watching, and if it resurfaces will reopen this issue.

Thanks for your hard work on this library!
